src/bentoml/_internal/context.py

```diff
@@ -72,21 +73,38 @@ def mutablecopy(self) -> Metadata:
         """
 
 
-@attr.define
 class InferenceApiContext:
-    request: "RequestContext"
-    response: "ResponseContext"
-
-    def __init__(self, request: "RequestContext", response: "ResponseContext"):
-        self.request = request
-        self.response = response
+    def __init__(self) -> None:
+        self._request_var: contextvars.ContextVar[
+            InferenceApiContext.RequestContext
+        ] = contextvars.ContextVar("request")
+        self._response_var: contextvars.ContextVar[
+            InferenceApiContext.ResponseContext
+        ] = contextvars.ContextVar("response")
+        # A dictionary for storing global state shared by the process
+        self.state: dict[str, t.Any] = {}
+
+    @contextlib.contextmanager
+    def in_request(
```

Comment:
We probably want a separate context for every request, right? Does this not run into race conditions it multiple requests are being served at once?

Reply:
No, since the request and response use contextvars and are coroutine-local.